docs/conf.py

```diff
@@ -155,3 +159,48 @@
 # ----------------------------------------------------------------------------------
 
 plot_html_show_formats = False
+
+
+# ----------------------------------------------------------------------------------
+# Source code links
+# ----------------------------------------------------------------------------------
+
+
+def linkcode_resolve(domain, info):
+    if domain != "py":
+        return None
+
+    module_name = info["module"]
+    module = sys.modules.get(module_name)
+    if module is None or "qiskit" not in module_name:
+        return None
+
+    obj = module
+    for part in info["fullname"].split("."):
+        try:
+            obj = getattr(obj, part)
+        except AttributeError:
+            return None
+        is_valid_code_object = (
+            inspect.isclass(obj) or inspect.ismethod(obj) or inspect.isfunction(obj)
+        )
+        if not is_valid_code_object:
+            return None
```

Comment:
I'm thinking we should just drop this logic altogether. Could we just use `importlib` to import the object directly? At the end of the day this is just trying to get the fully qualified path without doing an import at all. I also worry about manually accessing `sys.modules` directly

Reply:
I'll try switching to importlib. This current code is taken from how numpy does it and we didn't try importlib.

Reply:
@mtreinish is this better?